modules/management-console-agent/src/main/java/org/apache/ignite/agent/ws/WebSocketManager.java

```diff
@@ -259,136 +246,105 @@ private SslContextFactory createServerSslFactory(IgniteLogger log, ManagementCon
+
+        if (!F.isEmpty(cfg.getConsoleKeyStore()))
+            sslCtxConfigurator.setKeyStoreFile(cfg.getConsoleKeyStore());
+
+        if (!F.isEmpty(cfg.getConsoleKeyStorePassword()))
+            sslCtxConfigurator.setKeyStorePassword(cfg.getConsoleKeyStorePassword());
+
+        SslEngineConfigurator sslEngineConfigurator = trustAll
+            ? createTrustAllSslEngineConfigurator()
+            : new SslEngineConfigurator(sslCtxConfigurator, true, isNeedClientAuth, false);
+
+        if (!F.isEmpty(cfg.getCipherSuites()))
+            sslEngineConfigurator.setEnabledCipherSuites(cfg.getCipherSuites().toArray(EMPTY));
+
+        return sslEngineConfigurator;
     }
 
     /**
-     * @param content Key store content.
-     * @param pwd Password.
+     * @return SSL engine configurator with trust all manager and disabled hostname verification.
      */
-    private KeyStore keyStore(String content, String pwd) throws Exception {
-        KeyStore keyStore = KeyStore.getInstance("JKS");
+    private SslEngineConfigurator createTrustAllSslEngineConfigurator() {
+        try {
+            SSLContext ctx = SSLContext.getInstance("TLS");
+
+            ctx.init(null, new TrustManager[] {new DisabledX509TrustManager()}, null);
+
+            SslEngineConfigurator sslEngineConfigurator = new SslEngineConfigurator(ctx, true, false, false);
 
-        keyStore.load(new ByteArrayInputStream(content.getBytes(UTF_8)), pwd != null ? pwd.toCharArray() : null);
+            sslEngineConfigurator.setHostVerificationEnabled(false);
 
-        return keyStore;
+            return sslEngineConfigurator;
+        }
+        catch (NoSuchAlgorithmException | KeyManagementException e) {
+            throw new IgniteException(e);
+        }
     }
 
     /**
-     *
-     * @param keyStore Path to key store.
-     * @param keyStorePwd Optional key store password.
-     * @param trustAll Whether we should trust for self-signed certificate.
-     * @param trustStore Path to trust store.
-     * @param trustStorePwd Optional trust store password.
-     * @param ciphers Optional list of enabled cipher suites.
-     * @return SSL context factory.
+     * @param mgr Manager.
      */
-    @SuppressWarnings("deprecation")
-    private SslContextFactory sslContextFactory(
-            String keyStore,
-            String keyStorePwd,
-            boolean trustAll,
-            String trustStore,
-            String trustStorePwd,
-            List<String> ciphers
-    ) {
-        SslContextFactory sslCtxFactory = new SslContextFactory();
-
-        if (!F.isEmpty(keyStore)) {
-            try {
-                sslCtxFactory.setKeyStore(keyStore(keyStore, keyStorePwd));
-            }
-            catch (Exception e) {
-                log.warning("Failed to load server keyStore", e);
-            }
-        }
+    private void configureProxy(ClientManager mgr) {
+        String proxyHost = System.getProperty("http.proxyHost");
```

Comment:
What about https proxy? It's use `https.proxyHost` property

Reply:
Added.